<?xml version="1.0"?>
<launch>

<!--    <arg name="model" value="$(env ROBOT_MODEL)" />-->
    <arg name="model" value="omni" />
    <arg name="mission" default="M_2021-10-10_11-48-39__Field_16_1"/>
    <node pkg="rostopic" type="rostopic" name="pub_mission" args="pub /mission std_msgs/String $(arg mission) -r 1" output="screen" />

<!--    <node name="pub_odom" pkg="telemetry2mongo" type="gen_odom.py"/>-->
<!--    <node name="pub_camp" pkg="telemetry2mongo" type="pub_camp"/>-->
<!--    <node name="pub_fix" pkg="telemetry2mongo" type="pub_fix"/>-->

<!--    <node pkg="telemetry2mongo" type="multicore.sh" name="multicore_connection" output="screen" />-->

    <node pkg="rosbag" type="play" name="rosbag_play" args="$(find telemetry2mongo)/logger/logger_2021-02-26-15-26-20.bag -r 10 --start 0"/>

    <node name="telemetry2topic" pkg="telemetry2mongo" type="telemetry2topic">
        <param name="frequency" value="1" />
        <param name="odom_topic" value="/logger_odom_slam" />
        <param name="gps_topic" value="/logger_gps_slam_fix" />
        <param name="mission_topic" value="/mission" />
        <param name="payload_topic" value="/payload" />
        <param name="activation_service" value="/telemetry2topic/active" />
        <param name="platform" value="$(arg model)" />
        <param name="threshold_time" value="1.5" />
    </node>

</launch>
